### ISSUE TITLE
BatchNorm Issue

### ISSUE THREAD

**Issue by econti** [apache/incubator-mxnet#15644]
mxnet's `BatchNorm` doesn't seem to modify my input when used which is incorrect. Could someone take a look at the code below?

```
>>> from mxnet import gluon, nd
>>> import mxnet as mx
>>> import numpy as np
>>> def mx_DNN():
...     mx_DNN = gluon.nn.Sequential()
...     with mx_DNN.name_scope():
...         mx_DNN.add(
...             gluon.nn.BatchNorm(momentum=0.1, epsilon=1e-5, center=True, scale=True)
...         )
...     mx_DNN.collect_params().initialize(mx.init.Normal(0.02))
...     return mx_DNN
...
>>> mxnet_DNN = mx_DNN()
>>> data = nd.array([[1.5, 3.0], [-2.1, 2.1]])
>>> mxnet_DNN(data)

[[ 1.4999925  2.999985 ]
 [-2.0999894  2.0999894]]
<NDArray 2x2 @cpu(0)>
```

Environment:
```
Python 3.7.3
[GCC 7.3.0] :: Anaconda, Inc. on linux
mxnet-cu92==1.4.1
```

Thanks!

**Comment by wkcn** [comment]
The default mode is testing, so the mean is 0, and the std is 1.

The following code is in the training mode.

with mx.autograd.record():
  mxnet_DNN(data)

**Comment by zachgk** [comment]
@econti Did wkcn help you resolve your problem? If so, please close the issue. Thanks

**Comment by econti** [comment]
Yes, thank you @wkcn!
